### PR TITLE
EVG-7430: Color buildlogger lines

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

File: service/task.go [hunk 1]
```diff
@@ -889,3 +874,76 @@ func (uis *UIServer) testLog(w http.ResponseWriter, r *http.Request) {
...
+		}
+
+		lines = append(lines, line)
+	}
+
+	return lines
+}
+
+func readBuildloggerToChan(taskID string, r io.ReadCloser, lines chan apimodels.BuildloggerLogLine) {
+	var (
```
Comment (tychoish) [inline service/task.go]: I'd feel better if this took a context and returned a receive only channel

File: service/task.go [hunk 2]
```diff
@@ -889,3 +874,76 @@ func (uis *UIServer) testLog(w http.ResponseWriter, r *http.Request) {
...
+			if err != nil {
+				grip.Error(message.WrapError(err, message.Fields{
+					"task_id": taskID,
+					"message": "problem reading buildlogger log line severity",
+				}))
+				err = nil
+			}
+		}
+
+		lines <- apimodels.BuildloggerLogLine{
```
Comment (tychoish) [inline service/task.go]: send against context

File: service/task.go (hunk 1 above)
Comment (julianedwards) [reply service/task.go]: you mean send only, right?

File: service/task.go [hunk 3]
```diff
@@ -524,8 +525,16 @@ func (uis *UIServer) taskLog(w http.ResponseWriter, r *http.Request) {
 		var logReader io.ReadCloser
 		logReader, err = uis.getBuildloggerLogs(r.Context(), projCtx, r, logType, DefaultLogMessages, execution)
 		if err == nil {
-			gimlet.WriteText(w, logReader)
-			grip.Warning(logReader.Close())
+			defer func() {
+				err := logReader.Close()
+				if err != nil {
```
Comment (tychoish) [inline service/task.go]: you don't need to if-err because the conditional logging, so this can all be in one statment

File: service/task.go [hunk 4]
```diff
@@ -889,3 +886,84 @@ func (uis *UIServer) testLog(w http.ResponseWriter, r *http.Request) {
...
 	}
 }
+
+func readBuildloggerToSlice(ctx context.Context, taskID string, r io.ReadCloser) []apimodels.BuildloggerLogLine {
+	lines := []apimodels.BuildloggerLogLine{}
+	lineChan := make(chan apimodels.BuildloggerLogLine, 1024)
+	go readBuildloggerToChan(ctx, taskID, r, lineChan)
+
+	for {
+		line, more := <-lineChan
```
Comment (tychoish) [inline service/task.go]: you might do this in a select statment with a default

File: service/task.go [hunk 5]
```diff
@@ -889,3 +886,84 @@ func (uis *UIServer) testLog(w http.ResponseWriter, r *http.Request) {
...
+		case <-ctx.Done():
+			grip.Error(message.WrapError(ctx.Err(), message.Fields{
+				"task_id": taskID,
+				"message": "context error while reading buildlogger log lines",
+			}))
+		case lines <- apimodels.BuildloggerLogLine{
+			Message:  strings.TrimSuffix(line, "\n"),
+			Severity: getSeverityMapping(severity),
+		}:
+			continue
```
Comment (tychoish) [inline service/task.go]: this is redundant

File: service/task.go (hunk 4 above)
Comment (julianedwards) [reply service/task.go]: what would the default do?

**Pull request merged**